app/controllers/form_post_controller.rb

```diff
@@ -0,0 +1,31 @@
+class FormPostController < ApplicationController
+  def new
+    @current_user = current_user
+    @post = Post.new
+    respond_to do |format|
+      format.html { render :new, locals: { post: @post } }
+    end
+  end
+
+  def create
+    @current_user = current_user
+    @post = Post.new(post_params)
+    @post.author_id = @current_user.id
+    @post.CommentsCounter = 0
+    @post.LikesCounter = 0
+
+    if @post.save
+      flash[:success] = 'Post was successfully created.'
+      redirect_to users_path
+    else
+      flash.now[:alert] = 'Error: Post could not be created.'
+      render :new
+    end
+  end
+
+  private
+
+  def post_params
+    params.require(:post).permit(:Title, :Text)
+  end
+end
```

Comment:
**OPTIONAL**

- [ ] Please rename your file to `posts_controller.rb` and class `PostController` for easy readability
    - Remember to update the file name changes in your codebase with the new file names.